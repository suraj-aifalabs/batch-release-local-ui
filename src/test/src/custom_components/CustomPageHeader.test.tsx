import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import CustomPageHeaderTitle from '@/custom_components/CustomPageHeader';
import { MemoryRouter } from 'react-router-dom';

describe('CustomPageHeaderTitle component', () => {
    const testProps = {
        title: "Test Title",
        link: "/test-link"
    };

    it('renders title correctly', () => {
        render(
            <MemoryRouter>
                <CustomPageHeaderTitle {...testProps} />
            </MemoryRouter>
        );
        expect(screen.getByText(testProps.title)).toBeInTheDocument();
    });

    it('renders back arrow icon with correct link', () => {
        render(
            <MemoryRouter>
                <CustomPageHeaderTitle {...testProps} />
            </MemoryRouter>
        );
        const link = screen.getByRole('link');
        expect(link).toHaveAttribute('href', testProps.link);
        expect(link.querySelector('svg')).toBeInTheDocument();
    });

    it('applies correct styling to title', () => {
        render(
            <MemoryRouter>
                <CustomPageHeaderTitle {...testProps} />
            </MemoryRouter>
        );
        const title = screen.getByText(testProps.title);
        expect(title).toHaveClass('text-2xl');
        expect(title).toHaveClass('text-[#464545]');
        expect(title).toHaveClass('font-bold');
    });

    it('applies correct styling to back button', () => {
        render(
            <MemoryRouter>
                <CustomPageHeaderTitle {...testProps} />
            </MemoryRouter>
        );
        const icon = screen.getByRole('link').querySelector('svg');
        expect(icon).toHaveClass('w-8');
        expect(icon).toHaveClass('h-8');
        expect(icon).toHaveClass('p-1');
        expect(icon).toHaveClass('text-[#464545]');
        expect(icon).toHaveClass('hover:bg-gray-100');
        expect(icon).toHaveClass('rounded-full');
    });

    it('matches snapshot', () => {
        const { container } = render(
            <MemoryRouter>
                <CustomPageHeaderTitle {...testProps} />
            </MemoryRouter>
        );
        expect(container).toMatchSnapshot();
    });

    it('renders with different props', () => {
        const newProps = {
            title: "New Title",
            link: "/new-link"
        };
        render(
            <MemoryRouter>
                <CustomPageHeaderTitle {...newProps} />
            </MemoryRouter>
        );
        expect(screen.getByText(newProps.title)).toBeInTheDocument();
        expect(screen.getByRole('link')).toHaveAttribute('href', newProps.link);
    });
});