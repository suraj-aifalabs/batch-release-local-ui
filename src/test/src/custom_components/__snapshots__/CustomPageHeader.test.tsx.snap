// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomPageHeaderTitle component matches snapshot 1`] = `
<div>
  <div
    class="flex items-center justify-center gap-2"
  >
    <a
      data-discover="true"
      href="/test-link"
    >
      <svg
        class="remixicon w-8 h-8 p-1 text-[#464545] hover:bg-gray-100 rounded-full"
        fill="currentColor"
        height="24"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M10.8284 12.0007L15.7782 16.9504L14.364 18.3646L8 12.0007L14.364 5.63672L15.7782 7.05093L10.8284 12.0007Z"
        />
      </svg>
    </a>
    <div
      class="text-2xl text-[#464545] font-bold"
    >
      Test Title
    </div>
  </div>
</div>
`;
